src/ray/object_manager/common.cc

```diff
@@ -59,36 +51,32 @@ void PrintPlasmaObjectHeader(const PlasmaObjectHeader *header) {
                  << "metadata_size: " << header->metadata_size << "\n";
 }
 
-Status PlasmaObjectHeader::TryAcquireWriterMutex() {
-  // Try to acquire the lock, checking every 1s for the error bit.
-  struct timespec ts;
-  do {
-    if (has_error) {
-      return Status::IOError("channel closed");
-    }
-    clock_gettime(CLOCK_REALTIME, &ts);
-    ts.tv_sec += 1;
-  } while (pthread_mutex_timedlock(&wr_mut, &ts));
+Status PlasmaObjectHeader::TryToAcquireSemaphore(sem_t *sem) const {
+  if (has_error.load(std::memory_order_acquire)) {
+    return Status::IOError("channel closed");
+  }
+  RAY_CHECK_EQ(sem_wait(sem), 0);
```

Comment:
Hmm it will be an issue that we can't do a timed wait. We poll here because it's possible for the `has_error` bit to get set while the lock is being held by someone else.